File: z2_7/SerializeJson.cs
using System;
using System.Collections.Generic;
using System.IO;
using Newtonsoft.Json;

class SerializeJson
{
    public static void Run(DeserializeJson deserializedData, string fileLocation)
    {
        Console.WriteLine("Serializowanie JSON...");
        List<Dictionary<string, object>> processedData = new List<Dictionary<string, object>>();

        foreach (var item in deserializedData.Data)
        {
            processedData.Add(new Dictionary<string, object>
            {
                { "Kod_TERYT", item["Kod_TERYT"] },
                { "Województwo", item["Województwo"] },
                { "Powiat", item.ContainsKey("Powiat") ? item["Powiat"] : "Brak danych" },
                { "typ_JST", item["typ_JST"] },
                { "nazwa_urzędu_JST", item["nazwa_urzędu_JST"] },
                { "miejscowość", item["miejscowość"] },
                { "telefon_z_numerem_kierunkowym", item.ContainsKey("telefon_z_numerem_kierunkowym") ? item["telefon_z_numerem_kierunkowym"] : "Brak danych" }
            });
        }

        string jsonOutput = JsonConvert.SerializeObject(new { departaments = processedData }, Formatting.Indented);
        File.WriteAllText(fileLocation, jsonOutput);
        Console.WriteLine("Zapisano plik JSON!");
    }
}

File: lab2/serialize_json.py
# -*- coding: utf-8 -*-
"""
serialize json
"""
import json

class SerializeJson:
    # metoda statyczna
    @staticmethod
    def run(deserializeddata, filelocation):
        print("let's serialize something")
        lst = []
        for dep in deserializeddata.data:
            lst.append({
                "Kod_TERYT": dep['Kod_TERYT'],
                "Województwo": dep['Województwo'],
                "Powiat": dep.get('Powiat'),
                "typ_JST": dep['typ_JST'],
                "nazwa_urzędu_JST": dep['nazwa_urzędu_JST'],
                "miejscowość": dep['miejscowość'],
                "telefon_z_numerem_kierunkowym": "+" + str(dep.get('numer_kierunkowy', '')) + " " + str(dep.get('telefon'))
            })
        jsontemp = {"departaments": lst}
        with open(filelocation, 'w', encoding='utf-8') as f:
            json.dump(jsontemp, f, ensure_ascii=False, indent=4)